File: services_python/middlewares/cors.py
import os
from starlette.middleware.cors import CORSMiddleware


def add_cors_middleware(app):
    ALLOW_ORIGIN = os.getenv("ALLOW_ORIGIN", "http://localhost:5173")
    app.add_middleware(
        CORSMiddleware,
        allow_origins=[ALLOW_ORIGIN],
        allow_credentials=True,
        allow_methods=["GET", "HEAD", "OPTIONS", "PUT", "PATCH", "POST", "DELETE"],
        allow_headers=[
            "Origin",
            "Content-Length",
            "Content-Type",
            "Access-Control-Allow-Headers",
            "Authorization",
            "Set-Cookie",
        ],
    )

File: services_go/utils/cors.go
package utils

import (
	"github.com/gin-contrib/cors"
	"github.com/gin-gonic/gin"
)

var ALLOW_ORIGIN = GetEnv("ALLOW_ORIGIN", "http://localhost:5173")

func ConfigCORSMiddleware() gin.HandlerFunc {
	// Cấu hình middleware CORS
	config := cors.DefaultConfig()
	config.AllowAllOrigins = false
	config.AllowOrigins = append(config.AllowOrigins, ALLOW_ORIGIN)
	// config.AllowOrigins = []string{ALLOW_ORIGIN}
	config.AllowMethods = []string{"GET", "HEAD", "OPTIONS", "PUT", "PATCH", "POST", "DELETE"}
	config.AllowHeaders = []string{"Origin", "Content-Length", "Content-Type", "Access-Control-Allow-Headers", "Authorization", "Set-Cookie"}
	config.AllowCredentials = true
	return cors.New(config)
}
